x-pack/plugin/mapper-match-only-text/src/test/java/org/elasticsearch/xpack/matchonlytext/query/SourceConfirmedTextQueryTests.java

```diff
@@ -0,0 +1,325 @@
+                    .build();
+
+                Query sourceConfirmedPhraseQuery = new SourceConfirmedTextQuery(query, SOURCE_FETCHER_PROVIDER, Lucene.STANDARD_ANALYZER);
+
+                assertEquals(searcher.count(query), searcher.count(sourceConfirmedPhraseQuery));
+
+                ScoreDoc[] phraseHits = searcher.search(query, 10).scoreDocs;
+                assertEquals(2, phraseHits.length);
+                ScoreDoc[] sourceConfirmedHits = searcher.search(sourceConfirmedPhraseQuery, 10).scoreDocs;
+                CheckHits.checkEqual(query, phraseHits, sourceConfirmedHits);
+                CheckHits.checkExplanations(sourceConfirmedPhraseQuery, "body", searcher);
+
+                // Sloppy multi phrase query
+                query = new MultiPhraseQuery.Builder().add(new Term[] { new Term("body", "a"), new Term("body", "b") }, 0)
+                    .add(new Term[] { new Term("body", "d") }, 1)
+                    .setSlop(1)
+                    .build();
+                sourceConfirmedPhraseQuery = new SourceConfirmedTextQuery(query, SOURCE_FETCHER_PROVIDER, Lucene.STANDARD_ANALYZER);
+                assertEquals(searcher.count(query), searcher.count(sourceConfirmedPhraseQuery));
+                phraseHits = searcher.search(query, 10).scoreDocs;
+                assertEquals(2, phraseHits.length);
+                sourceConfirmedHits = searcher.search(sourceConfirmedPhraseQuery, 10).scoreDocs;
+                CheckHits.checkEqual(query, phraseHits, sourceConfirmedHits);
+                CheckHits.checkExplanations(sourceConfirmedPhraseQuery, "body", searcher);
+
+                // Multi phrase query with no matches
+                query = new MultiPhraseQuery.Builder().add(new Term[] { new Term("body", "d"), new Term("body", "c") }, 0)
+                    .add(new Term[] { new Term("body", "a") }, 1)
+                    .build();
+                sourceConfirmedPhraseQuery = new SourceConfirmedTextQuery(query, SOURCE_FETCHER_PROVIDER, Lucene.STANDARD_ANALYZER);
+                assertEquals(searcher.count(query), searcher.count(sourceConfirmedPhraseQuery));
+                assertArrayEquals(new ScoreDoc[0], searcher.search(sourceConfirmedPhraseQuery, 10).scoreDocs);
+
+                // Multi phrase query with one missing term
+                query = new MultiPhraseQuery.Builder().add(new Term[] { new Term("body", "d"), new Term("body", "c") }, 0)
+                    .add(new Term[] { new Term("body", "e") }, 1)
+                    .build();
+                sourceConfirmedPhraseQuery = new SourceConfirmedTextQuery(query, SOURCE_FETCHER_PROVIDER, Lucene.STANDARD_ANALYZER);
+                assertEquals(searcher.count(query), searcher.count(sourceConfirmedPhraseQuery));
+                assertArrayEquals(new ScoreDoc[0], searcher.search(sourceConfirmedPhraseQuery, 10).scoreDocs);
+            }
+        }
+    }
+
+    public void testSpanNear() throws Exception {
```

Comment:
Does this work, given the TODO in SourceConfirmedTextQuery implementation above?

Reply:
Yes, the query just uses a MatchAllDocsQuery as an approximation, but the query "works".

Reply:
Hopefully we'll create better approximations for spans in a follow-up.